### PR TITLE
switch from essential to oneconfig and add a lot of features!

### PR DESCRIPTION
**Pull request by CatalizCS**
- Switched from Essential to OneConfig GUI
- Added fast break feature
- Added moveable profit and status HUD
- Added random start and stop time to scheduler
- Added improved and customizable failsafe acting
- Added support for custom failsafe messages
- Added random failsafe macro stop time
- Added toggle to leave after failsafe is triggered
- Added toggle to pause visitors macro if player is in a contest
- Added toggle for banwave checker
- Added toggle for highlighting rewarp position
- Fixed dirt and rotation check
- Fixed negative timer issue
- Improved HUD icons and deleted unused ones

### CODE REVIEW

File: farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java [hunk 1]
```diff
@@ -290,33 +290,35 @@ public void onTick() {
...
+                    updateKeys((FarmHelper.config.VerticalMacroType != SMacroEnum.PUMPKIN_MELON.ordinal()) &&
+                        (FarmHelper.config.SShapeMacroType != SMacroEnum.CACTUS.ordinal()) && shouldWalkForwards(), (FarmHelper.config.SShapeMacroType == SMacroEnum.CACTUS.ordinal()) && shouldPushBack(), false, true, true);
                 }
                 return;
             case SWITCH_START:
                 if (!waitForChangeDirection.isScheduled()) {
                     if (switchBackwardsDirection) {
                         LogUtils.debugFullLog("Continue backwards");
-                        updateKeys(false, true, prevState == State.RIGHT, prevState == State.LEFT, true);
+                        updateKeys(false, true, prevState == State.RIGHT, prevState == State.LEFT, false);
```
Comment (May2Beez) [inline farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java:308]: Change the attack to `true` to make it more "natural", most of the ppl still holds the attack button while changing the lanr

File: farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java [hunk 2]
```diff
@@ -290,33 +290,35 @@ public void onTick() {
...
                         return;
                     } else {
                         LogUtils.debugFullLog("Continue forwards");
-                        updateKeys(true, false, prevState == State.RIGHT, prevState == State.LEFT, true);
+                        updateKeys(true, false, prevState == State.RIGHT, prevState == State.LEFT, false);
```
Comment (May2Beez) [inline farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java:312]: same as above

File: farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java [hunk 3]
```diff
@@ -290,33 +290,35 @@ public void onTick() {
...
                         return;
                     }
                 }
             case SWITCH_MID:
                 LogUtils.debugLog("Middle of switch, keep going");
                 if (switchBackwardsDirection)
-                    updateKeys(false, true, false, false, true);
+                    updateKeys(false, true, false, false, false);
```
Comment (May2Beez) [inline farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java:319]: ^

File: farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java [hunk 4]
```diff
@@ -290,33 +290,35 @@ public void onTick() {
...
                 else
-                    updateKeys(true, false, false, false, true);
+                    updateKeys(true, false, false, false, false);
```
Comment (May2Beez) [inline farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java:321]: ^

File: farmhelper-mod/src/main/java/com/jelly/farmhelper/network/proxy/ProxyManager.java [hunk 5]
```diff
@@ -19,25 +19,30 @@
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
 
+
+
 public class ProxyManager {
     public static boolean isTesting = false;
 
+    public enum ProxyType {
```
Comment (May2Beez) [inline farmhelper-mod/src/main/java/com/jelly/farmhelper/network/proxy/ProxyManager.java:27]: You've added HTTP and HTTPS to the `ProxyType.java` but here you didn't add it for some reason? 🤔

**Review by May2Beez — CHANGES_REQUESTED** [review]

File: farmhelper-mod/src/main/java/com/jelly/farmhelper/network/proxy/ProxyManager.java (hunk 5 above)
Comment (CatalizCS) [reply farmhelper-mod/src/main/java/com/jelly/farmhelper/network/proxy/ProxyManager.java:27]: I plan to do it, is it okay if I leave it?

Comment (May2Beez) [reply farmhelper-mod/src/main/java/com/jelly/farmhelper/network/proxy/ProxyManager.java:27]: As long as it's not visible to user through GUI it's okey, I thought you forgot about it

Comment (CatalizCS) [reply farmhelper-mod/src/main/java/com/jelly/farmhelper/network/proxy/ProxyManager.java:27]: kay :D

**Review by May2Beez — COMMENTED** [review]
The rotation time is fixed

File: farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java (hunk 1 above)
Comment (May2Beez) [reply farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java:308]: @CatalizCS Check this also, ppl will complain (again) about this

Comment (CatalizCS) [reply farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java]: added toggle in latest commit

**Review by May2Beez — APPROVED** [review]
Good, let the game begin. Testers, assemble

**Pull request merged**